plutus-chain-index-core/src/Marconi/Index/ScriptTx.hs

```diff
@@ -0,0 +1,194 @@
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+
+module Marconi.Index.ScriptTx where
+
+import Codec.Serialise (deserialiseOrFail)
+import Control.Lens.Operators ((^.))
+import Data.ByteString qualified as BS
+import Data.Foldable (forM_, toList)
+import Data.Maybe (catMaybes, fromJust)
+import Database.SQLite.Simple qualified as SQL
+import Database.SQLite.Simple.FromField qualified as SQL
+import Database.SQLite.Simple.ToField qualified as SQL
+import GHC.Generics (Generic)
+
+import Cardano.Api (SlotNo)
+import Cardano.Api qualified as C
+import Cardano.Api.Shelley qualified as Shelley
+import Cardano.Ledger.Alonzo.Language qualified as Alonzo
+import Cardano.Ledger.Alonzo.Scripts qualified as Alonzo
+import Cardano.Ledger.Core qualified
+import Cardano.Ledger.Crypto qualified as LedgerCrypto
+import Cardano.Ledger.Keys qualified as LedgerShelley
+import Cardano.Ledger.Shelley.Scripts qualified as LedgerShelley
+import Cardano.Ledger.ShelleyMA.Timelocks qualified as Timelock
+
+import RewindableIndex.Index.VSqlite (SqliteIndex)
+import RewindableIndex.Index.VSqlite qualified as Ix
+
+
+newtype Depth = Depth Int
+
+newtype ScriptAddress = ScriptAddress Shelley.ScriptHash
+  deriving (Show, Eq)
```

Comment:
Why the newtype? Because there's no `Show` and `Eq` for `ScriptHash`?

Reply:
It's so to not have it be an orphan instance of the ToField/FromField classes.